sdk/nodejs/appservice/zMixins.ts

```diff
@@ -501,3 +501,264 @@ export function getResourceGroupNameAndLocation(
     const getResult = resourceGroupName.apply(n => core.getResourceGroup({ name: n }));
     return { resourceGroupName, location: getResult.location };
 }
+
+export type CallbackArgs<C extends Context<R>, E, R extends Result> = {
+    /**
+     * The Javascript function instance to use as the entrypoint for the Azure FunctionApp.  Either
+     * [callback] or [callbackFactory] must be provided.
+     */
+    callback?: Callback<C, E, R>;
+
+    /**
+     * The Javascript function instance that will be called to produce the function that is the
+     * entrypoint for the Azure FunctionApp. Either [callback] or [callbackFactory] must be
+     * provided.
+     *
+     * This form is useful when there is expensive initialization work that should only be executed
+     * once.  The factory-function will be invoked once when the final Azure FunctionApp module is
+     * loaded. It can run whatever code it needs, and will end by returning the actual function that
+     * the Azure will call into each time the FunctionApp it is is invoked.
+     */
+    callbackFactory?: CallbackFactory<C, E, R>;
+}
+
+export function serializeFunctionCallback<C extends Context<R>, E, R extends Result>(
+    args: CallbackArgs<C, E, R>): pulumi.Output<pulumi.runtime.SerializedFunction> {

```

Comment:
1. consider juts mkaing this return a promise.  it doesn't really need to be an Output.
2. does tihs need to be exported?

Reply:
Changed to Promise and marked as internal (needs to be accessed from other mixin files)